### DISCUSSION TITLE
Transforming the ACVs from one antenna refrence frame to another

### DISCUSSION THREAD

**Discussion by ParisaShafiei** [groops-devs/groops#216] [Ideas]
Dear developers, 

Here, I would like to share my idea about writing a program to transform the measured antenna patterns from one frame top another.

There is a short description of the issue that brought me with this idea:

In POD of the CubeSatellites, I have noticed that there is a problem in the defined antenna reference frame in Chamber calibration and the one is used in processing.

Indeed, before launching the satellite, manufacturers measured the antennas in an anechonic chamber with using a structural model of the satellite. Then, they defined the test coordinate systems as the anechonic chamber frame and measured the antenna patterns. As a result, the measured antenna patterns are reported in the ANTEX format. But, unfortunately this ANTEX file cannot be used in POD or any processing due to the following limiting factors:

1. The true defined antenna reference frame of the satellite (let's call it "True") is different with the one used in measuring the patterns (let's call this one as "TEST" ). For example for the x axis, we have:

    **X_ARF_True  = -X_ARF_Test**

2. The azimuth is measured from the X_ARF_Test axis and it is increased by moving towards Y_ARF_Test axis.

So, as I faced into this issue, I was thinking that maybe it would be a good idea to have a program to do transformations in the antenna reference frame. So, if this issue or anything similar repeats for another satellite, there would be an option to get the measured antenna patterns in the right way in GROOPS. 

For example, in my case, I just did niot use the ACVs in processing and used the estimated ones, but I could not find an option to correct the measured patterns it in GROOPS and get the plot showing the ground-based calibration in the right way. 

Thank you in advance and I am looking forward to hearing from you.

With kind regards, 
Parisa

**Comment by tmayerguerr** [comment]
Hi Parisa,

I don't think I've ever encountered a problem like this before. But if you want to implement such a feature, I would do it in [GnssAntennaDefinitionCreate](https://groops-devs.github.io/groops/html/GnssAntennaDefinitionCreate.html). You might define a class like `GnssAntennaDefintionListTransformAxis` or similar. Within this class you read another [antenna](https://groops-devs.github.io/groops/html/gnssAntennaDefintionListType.html) (e.g. fromFile) and then transform it to the new frame.

Best regards
Torsten
